### PR TITLE
fix: corepack cannot be cached

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes https://github.com/netlify/pillar-workflow/issues/1046

Creates the corepack folder inside the netlify cache directories

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/build-image/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update the [included software doc](../included_software.md) (if you updated included software) 📄
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: run-build-functions.sh [hunk 1]
```diff
@@ -1015,27 +1016,28 @@ install_go() {
...
+      source $HOME/.gimme/env/go$GIMME_GO_VERSION.linux.amd64.env
+    else
+      echo "Failed to install Go version '$GIMME_GO_VERSION'"
+      exit 1
+    fi
   fi
 
-  source $gimmeEnvFile
 }
+
```
Comment (kitop) [inline run-build-functions.sh]: Are these changes related to the corepack one?

Comment (danez) [reply run-build-functions.sh]: Seems to be the revert of #887. Merge conflict?

Comment (lukasholzer) [reply run-build-functions.sh]: oh yea this is a merge conflict 😬

Comment (lukasholzer) [reply run-build-functions.sh]: Just resolved them

**Review by danez — APPROVED** [review]
